cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -181,6 +182,8 @@ func Run(s *options.CMServer) error {
 	if err != nil {
 		return err
 	}
+	// add a uniquifier so that two processes on the same host don't accidentally both become active
+	id = id + "_" + string(uuid.NewUUID())
```

Comment:
This should probably be a utility f(n) hidden in locking interface, just so it's uniform across components.

Reply:
> This should probably be a utility f(n) hidden in locking interface, just so it's uniform across components.

I don't think the locking interface should be opinionated about the shape of the holderidentities sent to it.

Reply:
I don't feel strongly, but folks will essentially copy-paste this code in multiple locations.